### ISSUE TITLE
Lora worked great until relaunch, then distorted garbled mess

### ISSUE THREAD

**Issue by johnnyshand73-lang** [ace-step/ACE-Step-1.5#434]
Very strange Lora behavior occurring. Lora was trained with default settings on turbo model. Initially great. On relaunch two hours later, with same settings and no update...horrible garbled generations. This was on the official gradio app. 

Console shows no errors or issues.

Now all Lora's are a mess, none will work correctly on base, turbo or sft, no matter the settings, and the effect has even spread to a separate Pinokio install. 

Things I've tried:
Re-installed Nvidia studio driver (546.01).
Cleared temp and huggingface cache
Deleted uv.lock to reinstall dependencies.

Link to before and after mp3's  and the Lora used: https://drive.google.com/file/d/13cWgFkWHV97TPpcbHTgL_XzmG88PfsST/view?usp=drive_link

Spec: RTX 3060 12GB
Windows 10

Any help or suggestions would be greatly appreciated.

Console:
r'.2026-02-11 01:41:02.793 | INFO     | acestep.handler:load_lora:212 - Restored base decoder before loading new LoRA
2026-02-11 01:41:02.793 | INFO     | acestep.handler:load_lora:215 - Loading LoRA adapter from F:\Ace\ACE-Step-1.5\lora_output\old2\checkpoints\epoch_500\adapter
2026-02-11 01:41:04.311 | INFO     | acestep.handler:load_lora:227 - LoRA adapter loaded successfully from F:\Ace\ACE-Step-1.5\lora_output\old2\checkpoints\epoch_500\adapter
2026-02-11 01:43:27.752 | INFO     | acestep.inference:generate_music:374 - [generate_music] LLM usage decision: thinking=False, use_cot_caption=False, use_cot_language=False, use_cot_metas=False, need_lm_for_cot=False, llm_initialized=False, use_lm=False
2026-02-11 01:43:27.752 | INFO     | acestep.gpu_config:get_gpu_memory_gb:168 - CUDA GPU detected: NVIDIA GeForce RTX 3060 (12.0 GB)
2026-02-11 01:43:27.752 | INFO     | acestep.handler:generate_music:3383 - [generate_music] Starting generation...
2026-02-11 01:43:27.752 | INFO     | acestep.handler:generate_music:3386 - [generate_music] Preparing inputs...
2026-02-11 01:43:27.869 | INFO     | acestep.handler:_load_model_context:981 - [_load_model_context] Loading vae to cuda2026-02-11 01:43:28.521 | INFO     | acestep.handler:_load_model_context:994 - [_load_model_context] Loaded vae to cuda in 0.6516s
2026-02-11 01:43:28.557 | INFO     | acestep.handler:_load_model_context:1000 - [_load_model_context] Offloading vae to CPU
2026-02-11 01:43:28.862 | INFO     | acestep.handler:_load_model_context:1014 - [_load_model_context] Offloaded vae to CPU in 0.3038s
2026-02-11 01:43:28.885 | INFO     | acestep.handler:_prepare_batch:2425 -
======================================================================
2026-02-11 01:43:28.885 | INFO     | acestep.handler:_prepare_batch:2426 - 🔍 [DEBUG] DiT TEXT ENCODER INPUT (Inference)2026-02-11 01:43:28.885 | INFO     | acestep.handler:_prepare_batch:2427 - ======================================================================
2026-02-11 01:43:28.887 | INFO     | acestep.handler:_prepare_batch:2428 - text_prompt:
# Instruction
Fill the audio semantic mask based on the given conditions:

# Caption
elv1. An intimate and melancholic ballad featuring a lone, fingerpicked nylon-string acoustic guitar and a deep, resonant male vocal. The performance is raw and emotional, with the singer's voice conveying a sense of longing and sadness. The arrangement is sparse, built around the intricate guitar work and the heartfelt vocal delivery.  The song includes a spoken-word passage over the guitar, enhancing its narrative quality.

# Metas
- bpm: 100
- timesignature: 4
- keyscale: N/A
- duration: 120 seconds
<|endoftext|>

2026-02-11 01:43:28.887 | INFO     | acestep.handler:_prepare_batch:2429 - ======================================================================

2026-02-11 01:43:28.909 | INFO     | acestep.handler:_load_model_context:981 - [_load_model_context] Loading vae to cuda2026-02-11 01:43:29.270 | INFO     | acestep.handler:_load_model_context:994 - [_load_model_context] Loaded vae to cuda in 0.3599s
2026-02-11 01:43:29.291 | INFO     | acestep.handler:_load_model_context:1000 - [_load_model_context] Offloading vae to CPU
2026-02-11 01:43:29.564 | INFO     | acestep.handler:_load_model_context:1014 - [_load_model_context] Offloaded vae to CPU in 0.2734s
2026-02-11 01:43:29.564 | INFO     | acestep.handler:preprocess_batch:2636 - [preprocess_batch] Inferring prompt embeddings...
2026-02-11 01:43:29.565 | INFO     | acestep.handler:_load_model_context:981 - [_load_model_context] Loading text_encoder to cuda
`loss_type=None` was set in the config but it is unrecognized. Using the default loss: `ForCausalLMLoss`.
2026-02-11 01:43:30.195 | INFO     | acestep.handler:_load_model_context:994 - [_load_model_context] Loaded text_encoder to cuda in 0.6294s
2026-02-11 01:43:31.434 | INFO     | acestep.handler:preprocess_batch:2639 - [preprocess_batch] Inferring lyric embeddings...
2026-02-11 01:43:31.434 | INFO     | acestep.handler:_load_model_context:1000 - [_load_model_context] Offloading text_encoder to CPU
2026-02-11 01:43:31.985 | INFO     | acestep.handler:_load_model_context:1014 - [_load_model_context] Offloaded text_encoder to CPU in 0.5498s
2026-02-11 01:43:31.986 | INFO     | acestep.handler:service_generate:2872 - [service_generate] Generating audio...
100%|█████████████████████████████████████████████████████████████████████████████| 100/100 [00:57<00:00,  1.74steps/s]
2026-02-11 01:44:29.887 | INFO     | acestep.handler:generate_music:3512 - [generate_music] Model generation completed. Decoding latents...
2026-02-11 01:44:30.132 | DEBUG    | acestep.handler:generate_music:3531 - [generate_music] pred_latents: torch.Size([1, 3000, 64]), dtype=torch.bfloat16
2026-02-11 01:44:30.132 | DEBUG    | acestep.handler:generate_music:3532 - [generate_music] time_costs: {'encoder_time_cost': 0.0839087963104248, 'diffusion_time_cost': 57.47817659378052, 'diffusion_per_step_time_cost': 0.5747817659378052, 'total_time_cost': 57.56208539009094, 'offload_time_cost': 2.7678542137145996}
2026-02-11 01:44:30.132 | INFO     | acestep.handler:generate_music:3549 - [generate_music] Decoding latents with VAE...2026-02-11 01:44:30.132 | INFO     | acestep.handler:_load_model_context:981 - [_load_model_context] Loading vae to cuda2026-02-11 01:44:30.492 | INFO     | acestep.handler:_load_model_context:994 - [_load_model_context] Loaded vae to cuda in 0.3599s
2026-02-11 01:44:30.503 | DEBUG    | acestep.handler:generate_music:3567 - [generate_music] Before VAE decode: allocated=7.81GB, max=10.50GB
2026-02-11 01:44:30.504 | INFO     | acestep.handler:generate_music:3581 - [generate_music] Using tiled VAE decode to reduce VRAM usage...
Decoding audio chunks: 100%|██████████████████████████████████████████████████████████| 7/7 [00:02<00:00,  3.25steps/s]
2026-02-11 01:44:32.989 | DEBUG    | acestep.handler:generate_music:3595 - [generate_music] After VAE decode: allocated=7.97GB, max=10.50GB
2026-02-11 01:44:33.035 | INFO     | acestep.handler:_load_model_context:1000 - [_load_model_context] Offloading vae to CPU
2026-02-11 01:44:33.335 | INFO     | acestep.handler:_load_model_context:1014 - [_load_model_context] Offloaded vae to CPU in 0.3001s
2026-02-11 01:44:33.335 | INFO     | acestep.handler:generate_music:3616 - [generate_music] VAE decode completed. Preparing audio tensors...
2026-02-11 01:44:33.335 | INFO     | acestep.handler:generate_music:3631 - [generate_music] Done! Generated 1 audio tensors.
2026-02-11 01:44:35.391 | DEBUG    | acestep.audio_utils:save_audio:125 - [AudioSaver] Fallback soundfile Saved audio to G:\ACE-Step-1.5\gradio_outputs\batch_1770774273\040ca6b0-85de-9384-4556-995da78e6d4e.mp3 (mp3, 48000Hz)
2026-02-11 01:44:36.207 | INFO     | acestep.gradio_ui.events.results_handlers:generate_with_batch_management:1676 - [generate_with_batch_management] Final yield: 46 core + 9 state

**Comment by ChuxiJ** [comment]
That’s weird. Text2Music works fine for you, but it only throws an error when loading LoRA, right?

Also, you’ve successfully loaded LoRA before, correct?
If you didn’t update the code and it stopped working after working previously, that suggests something is wrong with your environment, or there’s a small bug caused by an unintended change somewhere.

**Comment by johnnyshand73-lang** [comment]
Yes, Text2Music is fine without the Lora. And the Lora did work fine for over an hour of generations on base model with 100 steps. Then I closed the app for a while. The computer was offline. On relaunch it's as I said. The bizarre thing is that it is also affecting a separate install on Pinokio and seems system wide.

Here is a link to the before mp3 and after, as well as the lora: https://drive.google.com/file/d/13cWgFkWHV97TPpcbHTgL_XzmG88PfsST/view?usp=drive_link

**Comment by ChuxiJ** [comment]
Just to confirm: are you using the **base** model or the **SFT** model?
If you’ve trained a LoRA, we don’t recommend using the LM to generate codes (i.e., **thinking mode**).

**Comment by johnnyshand73-lang** [comment]
I'm using the base model with the Lora at 100 steps, with no thinking, Cot or LM.

**Comment by ChuxiJ** [comment]
It’s strange. The base model is not recommended for LoRA fine-tuning—it’s meant for full-parameter training instead.
If you fine-tune LoRA on the base model, a lot of logic may not be supported, such as CFG, etc. Correct behavior cannot be guaranteed here.

**Comment by johnnyshand73-lang** [comment]
I trained the Lora on the turbo model, but used it on the base model. If you listen to the files I uploaded to google drive, you'll hear the result is very good on the before mp3, then is horrible after relaunching.  I can't understand it.

**Comment by ChuxiJ** [comment]
The good results you got before might have been from the **turbo model**.
The model wasn’t successfully loaded as the **base model**. I’m not sure.

**Comment by johnnyshand73-lang** [comment]
Couldn't have been the turbo as I used 100 steps. Perhaps my environment is messed up because I have a few different versions as some were buggy, so I was downloading different ones to see if things were fixed, so I have the official gradio, the portable, and Pinokio, plus a couple of the forks, although I've never used them at the same time. 

Thank you for trying to help. I think I'll try a clean format, start from scratch. 
Fantastic application by the way!

**Comment by 1larity** [comment]
The build as of last night should be very stable and repeatable as far as generating from LoRA was concerned, I added a number of python unit tests, did countless user tests, successfully turning a Rock generation into Reggae (no mean feat) via LoRA. I did the tests on both a competent PC and an old box which should really not be capable of this kind of inference at all. 

In the end I did in excess of 20 review and fix cycles before I got no concerns from Copilot and Devstral, even coderabbit only had 2 very minor cases, which I'm not sur would _ever_ occur in practice, regardless, I fixed them too. I am _very_ confident adaptor selection is very robust and correct , a massive improvement over the previous implementation. 

There were a multitude of issues in loading the LoRA adaptor and scale mix-in in previous builds that made picking the adaptor non-deterministic. It's very likely in many scenarios,  no adapter was found at all, even though the user supplied one, and would silently fail, leaving the user believing LoRA was happening when it was in fact, not.

**Comment by ChuxiJ** [comment]
I've tested the LoRA and didn't find any major issues. It can train normally for now.

**Comment by ChuxiJ** [comment]
@1larity 

<img width="2520" height="1094" alt="Image" src="https://github.com/user-attachments/assets/29abb40a-30d0-4896-841a-7956309359c5" />

For me, this LoRA optimization has made training extremely slow.
It used to take only 1 hour, but now it takes 2 hours.
I'll adjust this part.
